src/server/api/routers/datasets.router.ts

```diff
@@ -0,0 +1,92 @@
+import { z } from "zod";
+import { createTRPCRouter, protectedProcedure, publicProcedure } from "~/server/api/trpc";
+import { prisma } from "~/server/db";
+import {
+  requireCanModifyDataset,
+  requireCanViewDataset,
+  requireNothing,
+} from "~/utils/accessControl";
+import userOrg from "~/server/utils/userOrg";
+
+export const datasetsRouter = createTRPCRouter({
+  list: protectedProcedure.query(async ({ ctx }) => {
+    // Anyone can list experiments
+    requireNothing(ctx);
+
+    const datasets = await prisma.dataset.findMany({
+      where: {
+        organization: {
+          organizationUsers: {
+            some: { userId: ctx.session.user.id },
+          },
+        },
+      },
+      orderBy: {
+        createdAt: "desc",
+      },
+      include: {
+        _count: {
+          // rename the field to numEntries
```

Comment:
?

Reply:
Haha forgot to take that out. It was an attempt to see what copilot would do